applications/FemToDemApplication/custom_processes/generate_dem_process.cpp

```diff
@@ -0,0 +1,170 @@
+                        distances(neigh) = dist_between_nodes;
+                        if (r_neighbour.GetValue(IS_DEM)) {
+                            has_dem_neigh = true;
+                            potential_radii(neigh) = dist_between_nodes - r_neighbour.GetValue(RADIUS);
+                            if (potential_radii(neigh) < 0.0 || potential_radii(neigh) / dist_between_nodes < 0.2) { // Houston-> We have a problem
+                                const double new_radius = dist_between_nodes*0.5;
+                                auto& r_radius_neigh_old = mrDEMModelPart.GetNode(r_neighbour.Id()).GetSolutionStepValue(RADIUS);
+                                r_radius_neigh_old = new_radius;
+                                r_neighbour.SetValue(RADIUS, new_radius);
+                                potential_radii(neigh) = new_radius;
+                            }
+                        } else {
+                            potential_radii(neigh) = 0.0;
+                        }
+                    }
+                    // Let's compute the Radius of the new DEM
+                    double radius;
+                    if (has_dem_neigh) {
+                        radius = this->GetMinimumValue(potential_radii);
+                    } else {
+                        radius = this->GetMinimumValue(distances)*0.5;
+                    }
+                    const array_1d<double,3>& r_coordinates = this->GetNodeCoordinates(r_node);
+                    this->CreateDEMParticle(r_node.Id(), r_coordinates, p_DEM_properties, radius, r_node);
+                }
+            }
+            it_elem->SetValue(DEM_GENERATED, true);
+            it_elem->Set(TO_ERASE, true);
+        } else if (!is_active && dem_generated)
+            it_elem->Set(TO_ERASE, true);
+    }
+}
+
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void GenerateDemProcess::CreateDEMParticle(
+    const int Id,
+    const array_1d<double, 3> Coordinates,
+    const Properties::Pointer pProperties,
+    const double Radius,
+    NodeType& rNode
+)
+{
+    mParticleCreator.CreateSphericParticle(mrDEMModelPart, Id, Coordinates, pProperties, Radius, "SphericParticle3D");
+    rNode.SetValue(IS_DEM, true);
+    rNode.SetValue(RADIUS, Radius);
+}
+
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+double GenerateDemProcess::CalculateDistanceBetweenNodes(
+    const NodeType& rNode1, 
+    const NodeType& rNode2
+    )
+{
+    const double X1 = rNode1.X();
+    const double X2 = rNode2.X();
+    const double Y1 = rNode1.Y();
+    const double Y2 = rNode2.Y();
+    const double Z1 = rNode1.Z();
+    const double Z2 = rNode2.Z();
```

Comment:
```suggestion
```

Reply:
in 3D i need to know the z components.. :)